cellxgene_schema_cli/scripts/gene_processing.py

```diff
@@ -209,9 +208,51 @@ def generate_gene_ref_diff(output_filename: str, current_ref_filepath: str, prev
                 removed_gene_ids.append(gene_id)
 
     diff_filepath = os.path.join(env.ONTOLOGY_DIR, f"{output_filename}_diff.txt")
-    with open(diff_filepath, "w") as f:
-        for gene_id in removed_gene_ids:
-            f.write(f"{gene_id}\n")
+    if removed_gene_ids:
+        with open(diff_filepath, "w") as f:
+            for gene_id in removed_gene_ids:
+                f.write(f"{gene_id}\n")
+    else:
+        print(f"No genes removed from {output_filename} reference. No diff created.")
+
+
+def process_gene_info(gene_info: dict) -> None:
+    """
+    Download the gene_info and convert it into a csv
+    :param gene_info: context to download and process the gene information.
+
+    """
+    print("Start", gene_info["description"])
+    # determine how to process based on file type
+    if gene_info["url"].endswith("gtf.gz"):
+        processer = _parse_gtf
+    elif gene_info["url"].endswith("txt"):
+        processer = _process_ercc
+    else:
+        raise TypeError(f"unknown file type: {gene_info['file_type']}")
+
+    # add version to URL if needed
+    url = gene_info["url"].format(version=gene_info["version"]) if gene_info.get("version") else gene_info["url"]
+
+    new_file = os.path.join(env.ONTOLOGY_DIR, f"new_genes_{gene_info['description']}.csv.gz")
```

Comment:
Rather then renaming the original file before proceeded, the new_file has a different name that does not collide with the original file. If everything goes will we replace the original is replace with the new_file.